File: static/taobao_product/主图/screenshot.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({
    viewport: { width: 850, height: 900 }
  });
  
  // 加载主图HTML
  await page.goto(`file://${process.cwd()}/主图生成器.html`);
  await page.waitForLoadState('networkidle');
  
  // 截取每个主图
  const images = await page.$$('.main-image');
  for (let i = 0; i < images.length; i++) {
    await images[i].screenshot({ path: `主图${i+1}.png` });
    console.log(`已保存: 主图${i+1}.png`);
  }
  
  await browser.close();
})();

File: static/taobao_product/主图/screenshot.py
import asyncio
from playwright.async_api import async_playwright
import os

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        
        # 截取主图
        page = await browser.new_page(viewport={'width': 850, 'height': 900})
        await page.goto(f'file://{os.getcwd()}/主图生成器.html')
        await page.wait_for_load_state('networkidle')
        await page.wait_for_timeout(1000)
        
        images = await page.query_selector_all('.main-image')
        for i, img in enumerate(images):
            await img.screenshot(path=f'主图{i+1}.png')
            print(f'已保存: 主图{i+1}.png')
        
        await browser.close()

asyncio.run(main())
